src/Native/src/runtime/type_serializer.cpp

```diff
@@ -82,6 +82,15 @@ result<datatype_t> deserialize_datatype_impl(TReader &sr) noexcept {
         auto uuid = sr.template read_unaligned<nncase::uuid_t>();
         auto size_bytes = sr.template read_unaligned<uint32_t>();
         return ok<datatype_t>(value_type_t(std::in_place, uuid, size_bytes));
+    }
+    case dt_vectortype: {
+        checked_try_var(elem_type, deserialize_datatype(sr));
+        auto rank = sr.template read_unaligned<int>();
```

Comment:
Size of `int` is platform dependent and may introduce bugs. It's better to use `uint8_t` as the rank of lanes is hardly larger than 255.